File: rps.py
#This is a python script that simulates a Rock Paper Scissors 
#with user input and the CPU making a random choice. The game is 
#intended to help with understanding conditional logic in the
#Python language.
#Author: Ja'Quis Franklin
#Date: 02/10/2025
import random

#Function to allow the computer to simulate an opponent
#Choosing between the options
def cpu_Choice():
    choices = ["Rock", "Paper", "Scissors"]
    choice = random.choice(choices)
    return choice

print("Let's Play Rock, Paper Scissors")
print("[R]ock")
print("[P]aper")
print("[S]cissors\n")

done = False
while not done:
    userChoice = input(str("SHOOT: "))

    match userChoice.lower().strip():
            case "r":
                userChoice = "Rock"
            case "p":
                userChoice = "Paper"
            case "s":
                userChoice = "Scissors"
            case _:
                print("Invalid. Choose Again:")
                continue

    cpuChoice = cpu_Choice()
    print(f"CPU chose: {cpuChoice}\n") 
    if userChoice == cpuChoice:
     print("Game Tied. Go Again!")
    elif (userChoice == "Rock" and cpuChoice == "Paper") or\
        (userChoice == "Scissors" and cpuChoice == "Rock") or\
        (userChoice == "Paper" and cpuChoice == "Scissors"):
        print("You Lose.")
        done = True
    else:
        print("YOU WIN!!")
        done = True

File: rps.c
/* This is a C program that plays rock paper scissors
   Author: Ja'Quis Franklin
   Date: 2/9/2025 */

#include <stdio.h>
#include <stdlib.h>
#include <time.h>
#include <stdbool.h>
#include <string.h>

const char *choice()
{
    const char *table[] = {"Rock", "Paper", "Scissors"};
    return table[rand() % 3];
}

int main()
{
    srand(time(NULL));

    printf("Welcome to the Rock Paper Scissors Game!!\n");
    printf("[R]ock\n");
    printf("[P]aper\n");
    printf("[S]cissors\n");

    bool end = false;
    while (!end)
    {
        char userInput[10];
        char userChoice[10];

        printf("\nSHOOT: ");
        fgets(userInput, sizeof(userInput), stdin);

        switch (userInput[0])
        {
        case 'r':
        case 'R':
            strcpy(userChoice, "Rock");
            break;
        case 'p':
        case 'P':
            strcpy(userChoice, "Paper");
            break;
        case 's':
        case 'S':
            strcpy(userChoice, "Scissors");
            break;
        default:
            printf("Invalid choice. Try again.\n");
            continue;
        }

        const char *cpuChoice = choice();
        printf("The CPU chose: %s\n", cpuChoice);

        if ((strcmp(cpuChoice, "Paper") == 0 && strcmp(userChoice, "Rock") == 0) ||
            (strcmp(cpuChoice, "Scissors") == 0 && strcmp(userChoice, "Paper") == 0) ||
            (strcmp(cpuChoice, "Rock") == 0 && strcmp(userChoice, "Scissors") == 0))
        {
            printf("You Lose.\n");
            end = true;
        }
        else if ((strcmp(cpuChoice, "Rock") == 0 && strcmp(userChoice, "Paper") == 0) ||
                 (strcmp(cpuChoice, "Paper") == 0 && strcmp(userChoice, "Scissors") == 0) ||
                 (strcmp(cpuChoice, "Scissors") == 0 && strcmp(userChoice, "Rock") == 0))
        {
            printf("You Win!!!\n");
            end = true;
        }
        else
        {
            printf("It's a tie! Choose again.\n");
        }
    }

    return 0;
}
